### PR TITLE
feat: add unit tests, refactor application and improve types

### PR DESCRIPTION
**Pull request by igolopolosov**
## Description:
Adding unit tests, refactoring application and improving types.

image

## Motivation:
The tests were missing since the start of the project, now it's good time to capture existing states of things before we move on implementing new features.

### CODE REVIEW

File: runner/providers/provider-github.ts [hunk 1]
```diff
@@ -42,14 +45,23 @@ export async function getRepository(project: Project) {
...
+) {
   const { organization } = await client.request(QUERY_GITHUB_REPOSITORIES, {
     organization: config.organization,
     amountOfRepos: config.amountOfRepos ?? DEFAULT_AMOUNT_OF_REPOS,
   })
-  const orgRepositories: string[] = organization.repositories.nodes.map(({ name }: { name: string }) => `${config.organization}/${name}`)
+  const orgRepositories: string[] = organization.repositories.nodes.map(
+    ({ name }: { name: string }) => `${config.organization}/${name}`
+  )
   const repos = config.repositories ?? orgRepositories ?? []
```
Comment (Laruxo) [inline runner/providers/provider-github.ts:61]: Early return `config.repositories` if it is defined (we don't need to make a request then). Also, I don't think `excludeRepos` should matter when using `repositories` option.

File: runner/providers/provider-github.ts [hunk 2]
```diff
@@ -42,14 +45,23 @@ export async function getRepository(project: Project) {
...
   const exclude = config.excludeRepos ?? []
 
-  return repos.filter(repo => exclude.every(exclusion => !repo.includes(exclusion)))
+  return repos.filter((repo) =>
```
Comment (Laruxo) [inline runner/providers/provider-github.ts:64]: I suggest adding `isArchived` to the organization query and filtering repos before doing any individual requests.

**Review by Laruxo — APPROVED** [review]
I am not the biggest fan of snapshot tests, but this looks okay.
Added a few comments about the logic.

File: runner/providers/provider-github.ts (hunk 2 above)
Comment (igolopolosov) [reply runner/providers/provider-github.ts:64]: It's already done, but in `scanProjects` function in https://github.com/monito/technical-stats/blob/main/runner/index.ts

**Pull request merged**